modules/web/js/dialog/new-item-dialog.js

```diff
@@ -31,6 +31,10 @@ class NewItemDialog extends ModalDialog {
 
     onSubmit(data, itemName) {
         const app = this._options.application;
+
+        if (!itemName.includes('.bal')) {
```

Comment:
Both the lowercase and uppercase extensions were considered.
 Fixed with commit 8287007